sklearn/datasets/openml.py

```diff
@@ -0,0 +1,415 @@
+        'feature_names', the original names of the dataset columns, and
+        'details' which provide more information on the openml meta-data.
+    """
+    data_home = get_data_home(data_home=data_home)
+    data_home = join(data_home, 'openml')
+    if cache:
+        mem = Memory(join(data_home, 'cache'), verbose=0).cache
+    else:
+        def mem(func):
+            return func
+    cached_get_data_info_by_name = mem(_get_data_info_by_name)
+    cached_get_data_description_by_id = mem(_get_data_description_by_id)
+    cached_get_data_features = mem(_get_data_features)
+    cached_download_data_arff = mem(_download_data_arff)
+
+    if not exists(data_home):
+        os.makedirs(data_home)
+
+    # check valid function arguments. data_id XOR (name, version) should be
+    # provided
+    if name is not None:
+        # OpenML is case-insensitive, but the caching mechanism is not
+        # convert all data names (str) to lower case
+        name = name.lower()
+        if data_id is not None:
+            raise ValueError(
+                "Dataset data_id={} and name={} passed, but you can only "
+                "specify a numeric data_id or a name, not "
+                "both.".format(data_id, name))
+        data_info = cached_get_data_info_by_name(name, version)
+        data_id = data_info['did']
+    elif data_id is not None:
+        # from the previous if statement, it is given that name is None
+        if version is not "active":
+            raise ValueError(
+                "Dataset data_id={} and version={} passed, but you can only "
+                "specify a numeric data_id or a version, not "
+                "both.".format(data_id, name))
+    else:
+        raise ValueError(
+            "Neither name nor data_id are provided. Please provide name or "
+            "data_id.")
+
+    data_description = cached_get_data_description_by_id(data_id)
+    if data_description['status'] != "active":
+        warn("Version {} of dataset {} is inactive, meaning that issues have"
+             " been found in the dataset. Try using a newer version.".format(
+                 data_description['version'], data_description['name']))
+
+    # download data features, meta-info about column types
+    features_list = cached_get_data_features(data_id)
+    features_dict = {feature['name']: feature for feature in features_list}
+
+    if target_column_name == "default-target":
+        target_column_name = _determine_default_target(features_list)
+
+    # TODO: stacking the content of the structured array
+    # this results in a copy. If the data was homogeneous
+    # and target at start or end, we could use a view instead.
+    data_columns = []
+    for feature in features_list:
+        # determine whether `feature` is a target
+        if (isinstance(target_column_name, string_types) and
+                feature['name'] == target_column_name):
+            is_target = True
+        elif (isinstance(target_column_name, list) and
```

Comment:
Can we make `target_column_name` always a list to simplify this code block?

I think we can just have 

```py
is_data = (feature['name'] not in target_columns and
           feature['is_ignore'] != 'false' and
           feature['is_row_identifier'] != 'false')
if is_data:
    ...
```

or even:

```py
data_columns = [feature for feature in features_list
                if (feature['name'] not in target_columns and
                    feature['is_ignore'] != 'false' and
                    feature['is_row_identifier'] != 'false')]
```

Reply:
Agreed.